README.md

```diff
@@ -51,15 +51,67 @@ This is the oldest rebus crossword from the New York Times (found by `grep -r Re
 
 The .xd format is a simple UTF-8 text file, and can often be 7-bit ASCII clean.
 
-Sections are delineated by two or more blank lines (3 consecutive newlines
-(0x0A)).  Subsections are delineated by a single blank line.
+The file is specified in one of two methods:
 
-### Headers (Section 1)
+- Using an implicit order, with sections being delineated by two or more blank 
+  lines (3 consecutive newlines (0x0A)).
+
+- Using `## [Section Name]` to declare the lines after as a certain section. 
+  Sections with case-insensitive headers which are not `"meta"`, `"grid"` or 
+  `"clues"` are ignored.  Order is unimportant.
+
+  <details>
+    <summary>An example of the previous full example using the explicit headers.</summary>
+
+      ## Meta
+
+      Title: New York Times, Saturday, January 1, 1955
+      Author: Anthony Morse
+      Editor: Margaret Farrar
+      Rebus: 1=HEART 2=DIAMOND 3=SPADE 4=CLUB
+      Date: 1955-01-01
+
+      ## Grid
+
+      1ACHE#ADAM#2LIL
+      BLUER#GULL#MATA
+      EATIN#APEX#ICER
+      ATAR#TILE#SNEAK
+      TEN#MANI#ITE###
+      ##DRUB#CANASTAS
+      FADED#BAGGY#OIL
+      ONES#KATES#TUNA
+      ETA#JOKER#JORUM
+      SILLABUB#SOON##
+      ###ACE#RUIN#ARK
+      3WORK#JINX#4MAN
+      BIRD#WADS#SCENE
+      ISLE#EDGE#PANEL
+      DEER#BEET#ARTEL
+
+      ## Clues
+
+      A1. Sadness. ~ HEARTACHE
+      A6. Progenitor. ~ ADAM
+      A10. Mae West stand-by. ~ DIAMONDLIL
+      [...]
+
+      D1. Vital throb. ~ HEARTBEAT
+      D2. Having wings. ~ ALATE
+      D3. Start the card game. ~ CUTANDDEAL
+      [...]
+
+  </details>
+
+
+HTML-style comments are allowed, started via `<!--`  and finishing via: `-->`.
```

Comment:
I don't think we should support HTML comments--too much of a pain to parse. There's the Notes section for whole-puzzle comments, and we could have a separate section for comments about individual clues.

Reply:
I thought it might be a pain, but unless you're giving errors with source locations attached, you can regex them out of the raw contents before you start the parser pretty safely.

That said, I'm happy to drop it from the spec though 👍🏻

Reply:
Yeah, I think with these items:

- remove HTML-style comments 
- change `Meta` section name to `Metadata`
- state that section names are case-insensitive
- the code in this repo (in particular the xdplayer package) should allow/parse .xd files with these new section names

then we can merge this PR into the repo.

Thanks for helping put this together!

Reply:
Cool, I've never written python, so I'll probably do the first three and see if I can persuade someone to do the implementation. I can document that this is a v2 of the spec, and that the python implementation only supports v1?

Reply:
okay, sounds good.  Thanks, @orta !